### ISSUE TITLE
TypeError: Cannot read property 'mode' of undefined

### ISSUE THREAD

**Issue by chadmccan** [chrisjshull/homebridge-nest#312]
I'm receiving this error in my log. I've enabled verbose and am running homebridge in debug mode. I don't see anything more descriptive. I don't see any Nest devices in my Home app. Any ideas?

```
[10/4/2020, 8:31:32 PM] [Nest] Authenticating via access token.
[10/4/2020, 8:31:33 PM] [Nest] Authentication successful.
[10/4/2020, 8:31:33 PM] [Nest] API subscribe POST: app launch
[10/4/2020, 8:31:34 PM] [Nest] REST API: structure 72662750-e18a-11e2-ad45-12313928905a, mounted 3 device(s)
[10/4/2020, 8:31:34 PM] [Nest] REST API: structure 780bf870-ec40-11e4-923b-22000b2712af, mounted 1 device(s)
[10/4/2020, 8:31:34 PM] [Nest] Nest API call to subscribe to device settings updates returned an error: TypeError: Cannot read property 'mode' of undefined
[10/4/2020, 8:31:34 PM] [Nest] Retrying in 10 seconds.
[10/4/2020, 8:32:00 PM] [Nest] API subscribe POST: subscribing
INFO: [Device Event]: (Garage Door Small) [TEMPERATURE] is 70.7
[10/4/2020, 8:34:00 PM] [Nest] API subscribe POST: subscribing
[10/4/2020, 8:36:00 PM] [Nest] API subscribe POST: subscribing
INFO: [Device Event]: (Garage Door Small) [TEMPERATURE] is 70.6
[10/4/2020, 8:38:00 PM] [Nest] API subscribe POST: subscribing
[10/4/2020, 8:40:01 PM] [Nest] API subscribe POST: subscribing
[10/4/2020, 8:42:01 PM] [Nest] API subscribe POST: subscribing
[10/4/2020, 8:42:03 PM] [Nest] Nest API call to subscribe to device settings updates returned an error: TypeError: Cannot read property 'mode' of undefined
[10/4/2020, 8:42:03 PM] [Nest] Retrying in 10 seconds.
[10/4/2020, 8:42:13 PM] [Nest] API subscribe POST: subscribing
[10/4/2020, 8:42:14 PM] [Nest] Nest API call to subscribe to device settings updates returned an error: TypeError: Cannot read property 'mode' of undefined
[10/4/2020, 8:42:14 PM] [Nest] Retrying in 10 seconds.
[10/4/2020, 8:42:24 PM] [Nest] API subscribe POST: subscribing
```

**Comment by adriancable** [comment]
@chadmccan - it looks like your Nest Thermostat doesn't have the Eco mode function. What kind of thermostat are you using? Is it a very old (generation 1) model? Approximately when did you buy it, and in what country?

I can put together a workaround for this, but more info on your thermostat type would be very useful.

**Comment by chadmccan** [comment]
Yeah, it is an older gen 1 device. Although, there is an Eco mode in the app. I’m In the US. Bought it in 2015.

**Comment by chadmccan** [comment]
I also have a Nest Protect, FYI. In case that may be causing issues too.

**Comment by adriancable** [comment]
@chadmccan - no, it's all the Thermostat. Can you send me an email at adrian.cable@gmail.com so I can give you some steps to collect some diagnostic info, which will help me code a workaround? Thanks!

**Comment by adriancable** [comment]
Was caused by a 1st generation thermostat without Eco mode. Fixed in 4.4.11.
